### PR TITLE
[DOCFIX] Add docs on debug logging for RPCs

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
note that, please only merge this PR to master.
this is not applicable to branch 2.2 before we cherrypick 
https://github.com/Alluxio/alluxio/pull/11073
https://github.com/Alluxio/alluxio/pull/11114
https://github.com/Alluxio/alluxio/pull/11146

File: docs/en/operation/Server-Logging.md [hunk 1]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+
+### Logging FUSE API calls
+
+Setting in `conf/log4j.properties`:
+
+```properties
+log4j.logger.alluxio.fuse.AlluxioFuseFileSystem=DEBUG
+```
+
+You will see debug logs on the begin and the end of each FUSE API call with its arguments and result
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
You will see debug logs at the beginning and the end of each FUSE API call with its arguments and result
```

File: docs/en/operation/Server-Logging.md [hunk 2]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+2020-03-03 14:33:35,132 DEBUG AlluxioFuseFileSystem - Enter: getattr(path=/aaa)
+2020-03-03 14:33:35,135 DEBUG AlluxioFuseFileSystem - Exit (0): getattr(path=/aaa) in 3 ms
+2020-03-03 14:33:35,138 DEBUG AlluxioFuseFileSystem - Enter: getattr(path=/._aaa)
+2020-03-03 14:33:35,140 DEBUG AlluxioFuseFileSystem - Failed to get info of /._aaa, path does not exist or is invalid
+2020-03-03 14:33:35,140 DEBUG AlluxioFuseFileSystem - Exit (-2): getattr(path=/._aaa) in 2 ms
+```
+
+### Logging RPCs Calls Sent by Client
+
+Add to your application-side `log4j.properties` to capture RPCs between Alluxio client
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
Add the following to your application-side `log4j.properties` to capture RPCs between the Alluxio client
```

File: docs/en/operation/Server-Logging.md [hunk 3]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+log4j.logger.alluxio.client.file.FileSystemMasterClient=DEBUG
+```
+
+Similarly, capture lower-level RPCs between Alluxio client and Block Master:
+
+```properties
+log4j.logger.alluxio.client.block.BlockMasterClient=DEBUG
+```
+
+You will see debug logs on the begin and the end of each RPC with its arguments and result
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
You will see debug logs at the beginning and end of each RPC with its arguments and result
```

File: docs/en/operation/Server-Logging.md [hunk 4]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+in client logs like the following:
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
in the client logs like the following:
```

File: docs/en/operation/Server-Logging.md [hunk 5]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+  syncIntervalMs: -1
+  ttl: -1
+  ttlAction: DELETE
+}
+) in 2 ms: alluxio.exception.status.NotFoundException: Path "/.DS_Store" does not exist.
+```
+
+### Logging RPC Calls Received by Masters
+
+On master, one can turn on the debug-level RPC logging for File System level RPC calls (e.g.,
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
On the master, debug-level RPC logging for File System level RPC calls can be turned on (e.g.,
```

File: docs/en/operation/Server-Logging.md [hunk 6]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+creating/reading/writing/removing files, updating file attributions) using `logLevel` command:
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
creating/reading/writing/removing files, updating file attributions) using the `logLevel` command:
```

File: docs/en/operation/Server-Logging.md [hunk 7]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+
+```console
+$ ./bin/alluxio logLevel \
+--logName=alluxio.master.file.FileSystemMasterClientServiceHandler \
+--target master --level=DEBUG
+```
+
+Similarly, turn on the debug-level logging for Block related RPC calls (e.g., adding/removing
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]:
```suggestion
Similarly, turn on the debug-level logging for block related RPC calls (e.g., adding/removing
```

File: docs/en/operation/Server-Logging.md [hunk 8]
```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
...
+--target master --level=DEBUG
+```
+
+### Identifying Expensive Client RPCs / FUSE calls
+
+When debugging the performance, it is often useful to understand which RPCs take most of the time
+but without recording all the communication (e.g., enabling all debug logging). There are two
+properties introduced in v2.3.0 in Alluxio to record expensive calls or RPCs in logs with WARN
+level. For example, setting the following in `conf/alluxio-site.properties` will record client-side
+RPCs taking more than 200ms and FUSE APIs taking more than 1s:
```
Comment (ZacBlanco) [inline docs/en/operation/Server-Logging.md]: Please put each sentence on a new line in this paragraph

Comment (apc999) [reply docs/en/operation/Server-Logging.md]: fixed

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**